src/services/github/v4/enablePullRequestAutoMerge.ts

```diff
@@ -0,0 +1,81 @@
+import { ValidConfigOptions } from '../../../options/options';
+import { apiRequestV4 } from './apiRequestV4';
+
+export interface PullRequestResponse {
+  repository: {
+    pullRequest: { id: string };
+  };
+}
+
+export interface PullRequestAutoMergeResponse {
+  enablePullRequestAutoMerge: { pullRequest: { number: number } };
+}
+
+export async function enablePullRequestAutoMerge(options: ValidConfigOptions) {
+  const {
+    accessToken,
+    githubApiBaseUrlV4,
+    pullNumber,
+    repoName,
+    repoOwner,
+    autoMergeMethod,
+  } = options;
+
+  if (!pullNumber) {
+    return;
+  }
+
+  const prQuery = /* GraphQL */ `
+    query GetRepoId(
+      $repoOwner: String!
+      $repoName: String!
+      $pullNumber: Int!
+    ) {
+      repository(owner: $repoOwner, name: $repoName) {
+        pullRequest(number: $pullNumber) {
+          id
+        }
+      }
+    }
+  `;
+
+  const prResponse = await apiRequestV4<PullRequestResponse>({
+    githubApiBaseUrlV4,
+    accessToken,
+    query: prQuery,
+    variables: {
+      repoOwner,
+      repoName,
+      pullNumber,
+    },
+  });
+
+  const pullRequestId = prResponse.repository.pullRequest.id;
+
+  const query = /* GraphQL */ `
+    mutation EnablePullRequestAutoMerge(
```

Comment:
Nice! I never tried mutations before. Always feel back to v3 api for those operations. This looks great though.

Reply:
Same here, I've never even used GraphQL before. This option isn't available in the REST api though

Reply:
Do you use https://github.com/graphql/graphiql ? I find it invaluable for exploring the API and trying out different queries.

Reply:
Yeah, I used GitHub's hosted version of it: https://docs.github.com/en/graphql/overview/explorer